runtime/codegen/automarshal.go

```diff
@@ -46,11 +40,20 @@ var (
 // calls). The registration is automatically done by generated code for custom
 // error structs that embed weaver.AutoMarshal.
 //
-// The extra PT type parameter is used to check that *T is an AutoMarshal
-// (since AutoMarshal methods have pointer receivers, but T itself may be a
-// struct).
-func RegisterSerializable[T any, PT autoMarshalPointer[T]]() {
+// REQUIRES: T can be serialized (either T or a pointer to T must implement
```

Comment:
I believe right now, we always generate `AutoMarshal` methods on `*T`, and I don't think users are supposed to implement `AutoMarshal` methods manually. Does it simplify things if we always require `*T` to implement `AutoMarshal`?

Reply:
That's a good point. Maybe I should abandon this change.

Reply:
Oh, maybe I misunderstood, but I thought most of the changes here were still needed to support errors where the `Error` method has a pointer receiver?

Reply:
You are correct. I was wondering if we need to support Error() with a pointer receiver.

Reply:
Ah, I'm in favor of supporting it.

Reply:
Ack. Just to clarify, yes the AutoMarshal methods are always defined on a pointer to a struct S.

The code here allows T to be either S or *S. This way, the user is free to place their Error() method on either S or *S.